src/feature/dirauth/process_descs.c

```diff
@@ -272,6 +331,14 @@ dirserv_router_get_status(const routerinfo_t *router, const char **msg,
         }
         return FP_REJECT;
       }
+
+      /* Check status with ed25519 key. */
+      r = dirserv_get_status_impl(
+              (char *) router->cache_info.signing_key_cert->signing_key.pubkey,
```

Comment:
dirserv_get_status_impl() does not accept ed25519 keys.

Reply:
nickm says: also, signing_key_cert can be null, so we shouldn't dereference it unconditionally